gpii/node_modules/flowManager/src/PCPChannel.js

```diff
@@ -0,0 +1,275 @@
+"use strict";
+
+
+var fluid = require("infusion");
+var gpii = fluid.registerNamespace("gpii");
+
+fluid.registerNamespace("gpii.flowManager");
+
+// A mixin grade applied to the lifecycleManager's session by the pcpChannel
+fluid.defaults("gpii.pcpChannel.sessionBinder", {
+    modelRelay: {
+        pcpChannel: {
+            source: { // Not "" because of FLUID-6192
+                segs: []
+            },
+            target: "{flowManager}.pcpChannel.model",
+            singleTransform: {
+                type: "gpii.pcpChannel.sessionToPCP",
+                ontologyMetadata: "{ontologyHandler}.ontologyMetadata"
+            },
+            // compensate for FLUID-6194
+            backward: "never",
+            forward: {
+                // avoid notifying the client for the init of LifecycleManagerSession, simplifies test and clients
+                excludeSource: "init"
+            }
+        }
+    },
+    modelListeners: {
+        updatePreferences: {
+            path: "{flowManager}.pcpChannel.model.settingControls",
+            includeSource: "PCP",
+            funcName: "gpii.pcpChannel.updatePreferences",
+            args: ["{flowManager}", "{pcpChannel}", "{flowManager}.lifecycleManager", "{change}.value"]
+        },
+        updateContext: {
+            path: "{flowManager}.pcpChannel.model.activeContextName",
+            includeSource: "PCP",
+            funcName: "gpii.pcpChannel.updateContext",
+            args: ["{flowManager}.contextManager", "{pcpChannel}", "{change}.value"]
+        }
+    }
+});
+
+fluid.defaults("gpii.pcpChannel", {
+    gradeNames: ["fluid.modelComponent"],
+    members: {
+        outputBlocked: null
+    },
+    distributeOptions: {
+        // Not {lifecycleManager session} because of FLUID-6193
+        target: "{flowManager lifecycleManager session}.options.gradeNames",
+        record: "gpii.pcpChannel.sessionBinder"
+    },
+    listeners: {
+        "{lifecycleManager}.events.onSessionStop": {
+            namespace: "pcpChannel",
+            funcName: "gpii.pcpChannel.sessionStop",
+            args: "{pcpChannel}"
+        }
+    }
+});
+
+gpii.pcpChannel.updatePreferences = function (flowManager, pcpChannel, lifecycleManager, newPrefs) {
```

Comment:
I remember that we discussed a while ago that the following scenario is possible:

- the user changes the value of a given setting
- the PSP notifies GPII about the modification
- GPII determines based on its internal rules that the new setting's value is incorrect
- GPII sends a message back to the PSP with the corrected setting's value (not necessarily the one prior to the user modification)

I do not see code related to the last point of the list in this PR. Will this be handled in a future improvement task?